kyo-core/shared/src/main/scala/kyo/Channel.scala

```diff
@@ -158,6 +158,54 @@ object Channel:
           */
         def full(using Frame): Boolean < (Abort[Closed] & IO) = IO.Unsafe(Abort.get(self.full()))
 
+        private def emitChunks(maxChunkSize: Maybe[Int] = Maybe.Absent)(using
+            Tag[A],
+            Frame
+        ): Ack < (Emit[Chunk[A]] & Abort[Closed] & Async) =
+            if maxChunkSize.exists(_ <= 0) then Ack.Stop
+            else if maxChunkSize.exists(_ == 1) then
+                Loop(()): _ =>
+                    Channel.take(self).map: v =>
+                        Emit.andMap(Chunk(v)):
+                            case Ack.Stop => Loop.done(Ack.Stop)
+                            case _        => Loop.continue(())
+            else
+                val drainEffect = maxChunkSize.fold(Channel.drain(self))(m => Channel.drainUpTo(self)(m - 1))
+                Loop[Unit, Ack, Abort[Closed] & Async & Emit[Chunk[A]]](()): _ =>
+                    Channel.take(self).map: a =>
+                        drainEffect.map: chunk =>
```

Comment:
I think this might not be very effective without batched inserts. Since channel writers are racing with this code, there's a risk that in most cases the emitted chunks will contain one or just a few values. I've created https://github.com/getkyo/kyo/issues/916 for batched inserts, we can address it as a follow up.

Reply:
When producing at <= rate of consumption, yes, but if the producer is faster than the consumer then chunking will be an optimization. If the producer is slower than the consumer, on the other hand, wouldn't we rather take small chunks quickly than wait for bigger chunks to be constructed?

Reply:
Good point, it's a good behavior for low latency. My concern is more about the race when the consumer is waiting for new elements. If that's the case and the producer has 10 items to add to the channel for example, as soon as the first element arrives, the consumer will wake up and try drain immediately without waiting for the producer to finish adding the 10 elements to the queue. A batched insert operation in `Channel` could reduce the impact of this race by waking up the consumer only after the 10 items are already in the queue.